CreditManagement/models.py

```diff
@@ -100,73 +99,417 @@ def __str__(self):
+    Created specially due to the different behaviour of the model (different database and model use)
+    """
+    def get_queryset(self, user=None, dining_list=None):
+        return DiningTransactionQuerySet.generate_queryset(user=user, dining_list=dining_list)
+
+    def annotate_users_balance(self, users, output_name=None):
+        return DiningTransactionQuerySet.annotate_user_balance(users=users, output_name=output_name)
+
+    def annotate_association_balance(self, associations, output_name=None):
+        return DiningTransactionQuerySet.annotate_association_balance(associations=associations, output_name=output_name)
+
+
+class PendingDiningTransaction(AbstractPendingTransaction):
     """
+    Model for the Pending Dining Transactions
+    Does NOT create a database, information is obtained elsewhere as specified in the manager/queryset
+    """
+    balance_annotation_name = "balance_pending_dining"
+    objects = PendingDiningTransactionManager()
+
     class Meta:
-        proxy = True
+        managed = False
+
+    @classmethod
+    def get_all_transactions(cls, user=None, association=None):
+        if association:
+            # Return none, no associations can be involved in dining lists
+            return cls.objects.none()
+        return cls.objects.get_queryset(user=user)
+
+    @classmethod
+    def get_user_balance(cls, user):
+        return cls.objects.all().compute_user_balance(user)
+
+    @classmethod
+    def get_association_balance(cls, association):
+        return cls.objects.all().compute_association_balance(association)
+
+    @classmethod
+    def annotate_balance(cls, users=None, associations=None, output_name=balance_annotation_name):
+        if associations:
+            return cls.objects.annotate_association_balance(associations, output_name=cls.balance_annotation_name)
+        else:
+            return cls.objects.annotate_users_balance(users, output_name=cls.balance_annotation_name)
+
+    @classmethod
+    def get_association_credit(cls, association):
+        """
+        Compute the balance according to this model based on the given association
+        :param association: The association
+        :return: The balance in Decimal
+        """
+        return Decimal(0.00)
 
-    objects = AssociationCreditQuerySet.as_manager()
+    def finalise(self):
+        raise NotImplementedError("PendingDiningTransactions are read-only")
 
-    def get_balance(self):
-        query = AssociationWithCredit.objects.filter(pk=self.pk).annotate_balance()
-        return query[0].balance
\ No newline at end of file
+    def __get_fixedform__(self):
+        return FixedTransaction(source_user=self.source_user, source_association=self.source_association,
+                                target_user=self.target_user, target_association=self.target_association,
+                                amount=self.amount,
+                                order_moment=self.order_moment, description=self.description)
+
+
```

Comment:
There is now a cyclic dependency between the apps `CreditManagement` and `Dining`. The `Dining` app uses models from `CreditManagement` and vice versa. It is cleaner to have only a dependency on `CreditManagement` from withing `Dining`, by moving the DiningList transaction models to the `Dining` app.

Reply:
See above, Creditmanagement now relies on dining, not the other way around